docs/framework/data/adonet/retrieving-data-using-a-datareader.md

```diff
@@ -1,131 +1,140 @@
-  
-> [!NOTE]
->  The Windows Server 2003 release of the .NET Framework includes an additional property for the **DataReader**, **HasRows**, which enables you to determine if the **DataReader** has returned any results before reading from it.  
-  
- The following code example iterates through a **DataReader** object, and returns two columns from each row.  
+ The following example iterates through a **DataReader** object and returns two columns from each row.  
   
  [!code-csharp[DataWorks SqlClient.HasRows#1](../../../../samples/snippets/csharp/VS_Snippets_ADO.NET/DataWorks SqlClient.HasRows/CS/source.cs#1)]
  [!code-vb[DataWorks SqlClient.HasRows#1](../../../../samples/snippets/visualbasic/VS_Snippets_ADO.NET/DataWorks SqlClient.HasRows/VB/source.vb#1)]  
   
- The **DataReader** provides an unbuffered stream of data that allows procedural logic to efficiently process results from a data source sequentially. The **DataReader** is a good choice when retrieving large amounts of data because the data is not cached in memory.  
-  
 ## Closing the DataReader  
- You should always call the **Close** method when you have finished using the **DataReader** object.  
+ Always call the **Close** method when you have finished using the **DataReader** object.  
   
- If your **Command** contains output parameters or return values, they will not be available until the **DataReader** is closed.  
+ If your **Command** contains output parameters or return values, those values are not available until the **DataReader** is closed.  
   
- Note that while a **DataReader** is open, the **Connection** is in use exclusively by that **DataReader**. You cannot execute any commands for the **Connection**, including creating another **DataReader**, until the original **DataReader** is closed.  
+ While a **DataReader** is open, the **Connection** is in use exclusively by that **DataReader**. You cannot execute any commands for the **Connection**, including creating another **DataReader**, until the original **DataReader** is closed.  
   
 > [!NOTE]
 >  Do not call **Close** or **Dispose** on a **Connection**, a **DataReader**, or any other managed object in the **Finalize** method of your class. In a finalizer, only release unmanaged resources that your class owns directly. If your class does not own any unmanaged resources, do not include a **Finalize** method in your class definition. For more information, see [Garbage Collection](../../../../docs/standard/garbage-collection/index.md).  
   
-## Retrieving Multiple Result Sets using NextResult  
- If multiple result sets are returned, the **DataReader** provides the **NextResult** method to iterate through the result sets in order. The following example shows the <xref:System.Data.SqlClient.SqlDataReader> processing the results of two SELECT statements using the <xref:System.Data.SqlClient.SqlCommand.ExecuteReader%2A> method.  
+## Retrieving Multiple Result Sets Using NextResult  
```

Comment:
Actually, our current convention is to use lowercase for everything other than the first word and proper nouns in a title and header. You can choose whether you want to do that in this topic, or leave title/headers as they are.